packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1933,15 +1942,19 @@ export class TransactionController extends BaseController<
       networkClientId: transactionMeta.networkClientId,
       chainId: transactionMeta.chainId,
     });
+    const provider = this.#multichainTrackingHelper.getProvider({
```

Comment:
Can we generate the above `EthQuery` instance using this `provider` to avoid multiple calls to the multichain helper?

Reply:
Done in https://github.com/MetaMask/core/pull/4055/commits/1e07ee78922768614afaf1db573c639d6ea068f4